include/jemalloc/internal/cache_bin.h

```diff
@@ -174,10 +180,26 @@ cache_bin_nonfast_aligned(const void *ptr) {
 	return ((uintptr_t)ptr & san_cache_bin_nonfast_mask) == 0;
 }
 
+/*
+ * If a cache bin was zero initialized (either because it lives in static or
+ * thread-local storage, or was memset to 0), this function indicates whether or
+ * not cache_bin_init was called on it.
+ */
+static inline
+bool cache_bin_still_zero_initialized(cache_bin_t *bin) {
+	return bin->stack_head == NULL;
+}
+
+static inline
+bool cache_bin_fake_initialized(cache_bin_t *bin) {
```

Comment:
Let's call this `cache_bin_disabled` and use it for the definite way of checking if the bin is enabled or not, regardless of ncached_max.

We detect disabled or not using a fake stack -- this is an impl detail and should not show up widely (the fake naming is confusing).